R/diffnet-class.r

```diff
@@ -563,42 +564,84 @@ new_diffnet <- function(graph, toa, t0=min(toa, na.rm = TRUE), t1=max(toa, na.rm
+      }
+    }
   }
 
   # Step 2.2: Checking names of toa
-  if (!length(names(toa)))
-    names(toa) <- meta$ids
+  if (num_of_behaviors==1) {
+    if (!length(names(toa))) {names(toa) <- meta$ids}
+  } else {
+    if (!length(rownames(toa))) { # Not necessary? toa_mat(toa, labels = meta$ids, t0=t0, t1=t1) already has labels
+      rownames(toa) <- meta$ids
+    }
+  }
 
   # Step 3.1: Creating Time of adoption matrix ---------------------------------
-  mat <- toa_mat(toa, labels = meta$ids, t0=t0, t1=t1)
+  mat <- toa_mat(toa, num_of_behaviors, labels = meta$ids, t0=t0, t1=t1)
 
   # Step 3.2: Verifying dimensions and fixing meta$pers
 
-  if (meta$type != "static") {
-    tdiff <- meta$nper - ncol(mat[[1]])
-    if (tdiff < 0)
-      stop("Range of -toa- is bigger than the number of slices in -graph- (",
-           ncol(mat[[1]]), " and ", length(graph) ," respectively). ",
-           "There must be at least as many slices as range of toa.")
-    else if (tdiff > 0)
-      stop("Range of -toa- is smaller than the number of slices in -graph- (",
-           ncol(mat[[1]]), " and ", length(graph) ," respectively). ",
-           "Please provide lower and upper boundaries for the values in -toa- ",
-           "using -t0- and -t- (see ?toa_mat).")
+  if (num_of_behaviors==1) {
+    if (meta$type != "static") {
+      tdiff <- meta$nper - ncol(mat$adopt)
+      if (tdiff < 0)
+        stop("Range of -toa- is bigger than the number of slices in -graph- (",
+             ncol(mat$adopt), " and ", length(graph) ," respectively). ",
+             "There must be at least as many slices as range of toa.")
+      else if (tdiff > 0)
+        stop("Range of -toa- is smaller than the number of slices in -graph- (",
+             ncol(mat$adopt), " and ", length(graph) ," respectively). ",
+             "Please provide lower and upper boundaries for the values in -toa- ",
+             "using -t0- and -t- (see ?toa_mat).")
+    } else {
+      graph <- lapply(1:ncol(mat$adopt), function(x) methods::as(graph, "dgCMatrix"))
+      meta  <- classify_graph(graph)
+    }
   } else {
-    graph <- lapply(1:ncol(mat[[1]]), function(x) methods::as(graph, "dgCMatrix"))
-    meta  <- classify_graph(graph)
+    if (meta$type != "static") {
+      tdiff <- meta$nper - ncol(mat[[1]]$adopt)
+      if (tdiff < 0)
+        stop("Range of -toa- is bigger than the number of slices in -graph- (",
+             ncol(mat[[1]]$adopt), " and ", length(graph) ," respectively). ",
+             "There must be at least as many slices as range of toa.")
+      else if (tdiff > 0)
+        stop("Range of -toa- is smaller than the number of slices in -graph- (",
+             ncol(mat[[1]]$adopt), " and ", length(graph) ," respectively). ",
+             "Please provide lower and upper boundaries for the values in -toa- ",
+             "using -t0- and -t- (see ?toa_mat).")
+    } else {
+
+      # This should be reviewed !! (here the graph becomes 'dynamic')
```

Comment:
?

Reply:
It took me a while realize that, at Step 1.1 in new_diffnet, a graph will never be 'static'. This is because, if I give as an input a single 'slice' of graph ('static'), that will be converted to a list of graph before this step anyway, so it become a 'dynamic' graph whatever the circumstance. This implies that the type of graph is shown as 'dynamic':
```
n <- 100
nper <- 5
static_graph <- rgraph_er(n = n, t = 1, p = 0.1) 

diffnet <- rdiffnet(
  seed.graph = static_graph,
  t = 5,
  seed.p.adopt = 0.1,
  rewire = FALSE,
  seed.nodes = 'central',
  threshold.dist = runif(n, .1, .4)
)

str(diffnet) # always meta$type == dinamic !!
```
Since the code never really uses those lines (neither single diff nor multi diff), it's "working."